Products/CMFPlone/resources/browser/resource.py

```diff
@@ -48,24 +47,24 @@ def _request_bundles(self):
             request_disabled_bundles.update(getattr(request, "disabled_bundles", []))
         return request_enabled_bundles, request_disabled_bundles
 
-    def _user_local_roles(self, site):
-        portal_membership = getToolByName(site, "portal_membership")
+    def _user_local_roles(self):
+        portal_membership = getToolByName(getSite(), "portal_membership")
         user = portal_membership.getAuthenticatedMember()
         return "|".join(user.getRolesInContext(self.context))
```

Comment:
```suggestion
        return "|".join(sorted(set(user.getRolesInContext(self.context))))
```

We get a list here, and it might contain duplicates (I did not see this in a quick test, but I guess it might happen), and we don't want different cache keys when the order is different.

I know this line did not change in this PR, but seems good to me to include this, because we will get different cache keys per context now, instead of per site.